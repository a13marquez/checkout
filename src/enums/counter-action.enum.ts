export enum CounterActionEnum {
  INCREMENT,
  DECREMENT,
}
