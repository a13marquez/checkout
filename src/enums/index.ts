export * from './product-code.enum';
export * from './counter-action.enum';
