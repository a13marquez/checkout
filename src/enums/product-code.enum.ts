export enum ProductCodeEnum {
  Berliner = 'BERLINER',
  Hoppy = 'HOPPY',
  Ipa = 'IPA',
}
