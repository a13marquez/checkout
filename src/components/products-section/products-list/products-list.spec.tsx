import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { ProductsList } from './index';
import { ProductCodeEnum } from '../../../enums';

describe('Section Component: ', () => {
  let productsList;
  let updateCounter = jest.fn((code, quantity) => ({
    code: ProductCodeEnum.Cap,
    name: 'Cabify Cap',
    price: 5.00,
    discount: true,
    bulkDiscount: false,
    quantity: 0,
    img: '../../../img/cap.png',
  }));
  let updateTotal = jest.fn((price, items) => ({price: 1, items: 1}))
  const products = [{
    code: ProductCodeEnum.Cap,
    name: 'Cabify Cap',
    price: 5.00,
    discount: true,
    bulkDiscount: false,
    quantity: 0,
    img: '../../../img/cap.png',
  }, {
    code: ProductCodeEnum.Tshirt,
    name: 'Cabify T-Shirt',
    price: 20.00,
    discount: false,
    bulkDiscount: true,
    bulkPrice: 19.00,
    bulkQuantity: 3,
    quantity: 0,
    img: '../../../img/shirt.png',
  }, {
    code: ProductCodeEnum.Mug,
    name: 'Cabify Coffe Mug',
    price: 7.50,
    discount: false,
    bulkDiscount: false,
    quantity: 0,
    img: '../../../img/mug.png',
  }];
  afterEach(() => {
    productsList = null;
  });
  test('It renders Product List component', () => {
    productsList = mount(<ProductsList
      products={[]}
      isFetching={false}
      updateCounter={updateCounter}
      updateTotal={updateTotal}
      fetchProducts={() => []}
      currency="USD"
    />);
    expect(productsList).toMatchSnapshot();
  });

  test('It renders Products List without props', () => {
    productsList = mount(<ProductsList
      products={[]}
      isFetching={false}
      updateCounter={updateCounter}
      updateTotal={updateTotal}
      fetchProducts={() => []}
      currency="USD"
    />);
    expect(productsList.find('li')).toHaveLength(0);
  });

  test('It renders Products List with props', () => {
    productsList = mount(<ProductsList
      products={products}
      isFetching={false}
      updateCounter={updateCounter}
      updateTotal={updateTotal}
      fetchProducts={() => products}
      currency='USD'
    />);
    expect(productsList.find('li')).toHaveLength(3);
  });
});
