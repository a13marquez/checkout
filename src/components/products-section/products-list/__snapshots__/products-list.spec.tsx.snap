// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component:  It renders Product List component 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  vertical-align: baseline;
  font: 100%;
  list-style: none;
  position: relative;
  width: 100%;
  padding-left: 0;
}

<ProductsList
  currency="USD"
  fetchProducts={[Function]}
  isFetching={false}
  products={Array []}
  updateCounter={[MockFunction]}
  updateTotal={[MockFunction]}
>
  <Styled(styled.ul)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  margin: 0;
  padding: 0;
  border: 0;
  vertical-align: baseline;
  font: 100%;
",
              "
  list-style: none;
",
              "
  position: relative;
  width: 100%;
  padding-left: 0;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "Styled(styled.ul)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <ul
        className="c0"
      />
    </StyledComponent>
  </Styled(styled.ul)>
</ProductsList>
`;
